cpp/src/merge/merge.cu

```diff
@@ -143,6 +145,12 @@ void materialize_bitmask(column_view const& left_col,
   CHECK_CUDA(stream.value());
 }
 
+struct side_index_generator {
+  side _side;
+
+  __device__ index_type operator()(size_type i) const noexcept { return index_type{_side, i}; }
```

Comment:
Does this lambda work? (with trailing-return-type)
`[] __device__(auto i)  -> index_type { return index_type{_side, i}; }`

Reply:
I'm not going to experiment with that since the functor is already made.